prometheus_hardware_exporter/collectors/redfish.py

```diff
@@ -56,23 +86,338 @@ def _retrieve_redfish_sensor_data(self) -> List[Any]:
         Returns:
             sensors: List of dicts with details for each sensor
         """
-        try:
-            with redfish.redfish_client(
-                base_url=self.host,
-                username=self.username,
-                password=self.password,
-                timeout=self.timeout,
-                max_retry=self.max_retry,
-            ) as redfish_obj:
-                return redfish_utilities.get_sensors(redfish_obj)
-        except (
-            InvalidCredentialsError,
-            SessionCreationError,
-            ConnectionError,
-            RetriesExhaustedError,
-        ) as err:
-            logger.exception(err)
-        return []
+        sensors: List[Any] = []
+
+        redfish_obj: Optional[HttpClient] = self._get_redfish_obj()
+        if redfish_obj:
+            logger.info("Getting redfish sensor info...")
+            sensors = redfish_utilities.get_sensors(redfish_obj)
+            redfish_obj.logout()
+        return sensors
+
+    def _verify_redfish_call(self, redfish_obj: HttpClient, uri: str) -> Optional[Dict[str, Any]]:
+        """Return REST response for GET request API call on the URI.
+
+        Returns None if URI isn't present.
+        """
+        resp: RestResponse = redfish_obj.get(uri)
+        if resp.status == 200:
+            return resp.dict
+        logger.info("Not able to query from URI: %s.", uri)
+        return None
+
+    def get_processor_data(self) -> Tuple[Dict[str, int], Dict[str, List]]:
+        """Return processor data and count.
+
+        Returns:
+            processor_count: Dict of system ids mapped to the total number
+            of processors on that system.
+            processor_data: Dict of system ids mapped to list of processors on that system.
```

Comment:
**suggestion:** 

We should provide the example of processor_data in doc-string.